src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1350,9 +1378,394 @@ public S(int arg)
 ");
             var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
             comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.Equal("System.Int32 S.<P>k__BackingField", comp.GetTypeByMetadataName("S").GetMembers().OfType<FieldSymbol>().Single().ToTestDisplayString());
+            comp.VerifyDiagnostics();
+
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void SpeculativeSemanticModel_FieldInRegularAccessor_BindOriginalFirst()
```

Comment:
>SpeculativeSemanticModel_FieldInRegularAccessor_BindOriginalFirst [](http://example.com/codeflow?start=20&length=65)

Another scenario that came to mind is Expression Evaluation, see ```Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.UnitTests.ExpressionCompilerTests.EvaluateThis``` unit-test for example.
I think we need to test an attempt to evaluate ```field``` within an accessor:
- of a regular property (doesn't use `field` keyword)
- of a property that uses `field` keyword.

Also, edit-and-continue (ENC) scenarios are interesting. Specifically when an edit introduces/eliminates usages of `field` keyword for a property. A PROTOTYPE comment for ENC tests should be good enough for this particular PR.